test/test_bitbox02.py

```diff
@@ -0,0 +1,109 @@
+
+        atexit.register(self.stop)
+
+    def stop(self):
+        super().stop()
+        self.simulator_proc.terminate()
+        self.simulator_proc.wait()
+        self.log.close()
+        atexit.unregister(self.stop)
+
+class TestBitbox02GetXpub(DeviceTestCase):
+    def test_getxpub(self):
+        self.dev_args.remove('--chain')
+        self.dev_args.remove('test')
+        result = self.do_command(self.dev_args + ['--expert', 'getxpub', 'm/84h/0h/0h/3'])
+        self.assertEqual(result['xpub'], 'xpub6F8W4c3nJf6vWyEQPW9rofRgKf9LUWrbLc6fh2GUgofxXzuMwNEXw9dUuAeHuNiu2MebTmLX1CY2wxN1pgUuQtsWa9x8QBk7J51nD86vann')
+        self.assertFalse(result['testnet'])
+        self.assertFalse(result['private'])
+        self.assertEqual(result['depth'], 4)
+        self.assertEqual(result['parent_fingerprint'], 'd934efde')
+        self.assertEqual(result['child_num'], 3)
+        self.assertEqual(result['chaincode'], '03b0d37df586659fb87145e1d28506e4e2d42777586568d61ecdf6c9e041a0a1')
+        self.assertEqual(result['pubkey'], '03290b94a942a317c3846244f1eb6d67214326c8cfc6d940c823ace57ab818dbbd')
+
+def bitbox02_test_suite(simulator, bitcoind, interface):
+    dev_emulator = BitBox02Emulator(simulator)
+
+    signtx_cases = [
+        (["segwit"], [], False, False)
+    ]
+
+    # Generic Device tests
+    suite = unittest.TestSuite()
+    suite.addTest(DeviceTestCase.parameterize(TestDeviceConnect, bitcoind, emulator=dev_emulator, interface=interface, detect_type=dev_emulator.type))
+    suite.addTest(DeviceTestCase.parameterize(TestGetKeypool, bitcoind, emulator=dev_emulator, interface=interface))
+    suite.addTest(DeviceTestCase.parameterize(TestGetDescriptors, bitcoind, emulator=dev_emulator, interface=interface))
+    suite.addTest(DeviceTestCase.parameterize(TestSignTx, bitcoind, emulator=dev_emulator, interface=interface, signtx_cases=signtx_cases))
+    suite.addTest(DeviceTestCase.parameterize(TestDisplayAddress, bitcoind, emulator=dev_emulator, interface=interface))
+    # TestSignMessage is removed, since its only testcase is for legacy p2pkh, which is not supported by BitBox02
+    # suite.addTest(DeviceTestCase.parameterize(TestSignMessage, bitcoind, emulator=dev_emulator, interface=interface))
+    suite.addTest(DeviceTestCase.parameterize(TestBitbox02GetXpub, bitcoind, emulator=dev_emulator, interface=interface))
+
+    result = unittest.TextTestRunner(stream=sys.stdout, verbosity=2).run(suite)
+    return result.wasSuccessful()
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Test BitBox02 implementation')
+    parser.add_argument('simulator', help='Path to simulator binary')
+    parser.add_argument('bitcoind', help='Path to bitcoind binary')
+    parser.add_argument('--interface', help='Which interface to send commands over', choices=['library', 'cli', 'bindist'], default='library')
+    args = parser.parse_args()
+
+    sys.exit(not bitbox02_test_suite(args.simulator, None, None))
```

Comment:
I had some problems with this test. It seems you should pass bitcoind and interface in `bitbox02_test_suite ` instead of None, no?

Reply:
Yes, you're right. Fixed that now.